### PR TITLE
Variant of RxHelper.unmarhsaller that can use a custom mapper

### LINKED ISSUE #34: Variant of RxHelper.unmarhsaller that can use a custom mapper
We use jackson-dataformat-yaml in order to unmarshall YAML content to a POJO. It would be very nice if the unmarshaller had a variant that allowed for sending in a mapper, and delegate that mapper to the relevant operator. 

I can give a stab at implementing this and sending a PR if that is ok. It should really not be that hard.

### PR DESCRIPTION
**Pull request by bjartek**
I added the method to the unmarhal helper we talked about ealier. The docs only mention the method with the concrete type and not the type reference, but I think that is enough. What do you think?

### CODE REVIEW

**Comment by vietj** [comment]
can you cover also RxJava2 as well ?

**Comment by bjartek** [comment]
Let me look into it. I am not that familiar with RxJava2. Any pointers are welcome :)

**Comment by bjartek** [comment]
Was this what you had in mind for RxJava2?

There is quite a lot of duplication here. I just added the ObjectMapper to all the unmarshal methods and fixed the underlying Operators in the same way.

**Comment by vietj** [comment]
can you add tests for RxJava2 ?

**Comment by bjartek** [comment]
Sorry for the huge delay here. Was swamped at work. I did not include a YamlFactory into the custom mapper. Do you want to me add that as a test dependency and convert all the json to yaml?

File: rx-groovy/src/main/asciidoc/groovy/index.adoc [hunk 1]
```diff
@@ -103,6 +103,16 @@ transforms an `Observable<Buffer>` in json format into an object observable:
 include::unmarshaller.groovy[tags=example]
 ----
 
+== Yaml unmarshalling
```
Comment (tsegismont) [inline rx-groovy/src/main/asciidoc/groovy/index.adoc]: Jackson alternative formats are not limited to Yaml (XML for sure, I believe there are others). Could you update this title to reflect the diversity of text backends?

File: rx-groovy/src/main/java/io/vertx/rx/groovy/RxHelper.java [hunk 2]
```diff
@@ -76,4 +77,50 @@ public static RxJavaSchedulersHook schedulerHook(Context context) {
       }
     };
   }
+
+  /**
+   * Returns a json unmarshaller for the specified java type as a {@link rx.Observable.Operator} instance.<p/>
```
Comment (tsegismont) [inline rx-groovy/src/main/java/io/vertx/rx/groovy/RxHelper.java]: IIUC it's not just json but any format the provided mapper is configured to work with

File: rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+
+/**
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
```
Comment (tsegismont) [inline rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java]: Can you put your name here?

File: rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java [hunk 4]
```diff
@@ -0,0 +1,264 @@
+package io.vertx.reactivex.test;
```
Comment (tsegismont) [inline rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java]: Can you add a license header please?

**Comment by tsegismont** [comment]
Fixes #34

**Comment by tsegismont** [comment]
>  I did not include a YamlFactory into the custom mapper. Do you want to me add that as a test dependency and convert all the json to yaml?

@bjartek  what do you mean?

File: rx-groovy/src/main/asciidoc/groovy/index.adoc (hunk 1 above)
Comment (bjartek) [reply rx-groovy/src/main/asciidoc/groovy/index.adoc]: Sure

File: rx-groovy/src/main/java/io/vertx/rx/groovy/RxHelper.java (hunk 2 above)
Comment (bjartek) [reply rx-groovy/src/main/java/io/vertx/rx/groovy/RxHelper.java]: Good point

File: rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java (hunk 4 above)
Comment (bjartek) [reply rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java]: No problem

File: rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java (hunk 3 above)
Comment (bjartek) [reply rx-java2/src/test/java/io/vertx/reactivex/test/BufferCustomMapperTest.java]: I would love to :)

**Comment by bjartek** [comment]
I have signed the Eclipse CLA.

**Comment by bjartek** [comment]
I have fixed most of you comments and even updated the branch against upstream. 

My comment from above was if you want the actual tests to try to use a custom mapper factory instead of the build in JSON one. In order to do that we would need to include a dataformat for another type as a test dependency and add some tests to show that it would work.

**Comment by tsegismont** [comment]
@bjartek sorry for the late reply. Yes, having yaml/csv or any other pluggable data format in the tests would be great

**Comment by bjartek** [comment]
@tsegismont sorry for the delay but here are the requests changes.

**Comment by tsegismont** [comment]
@bjartek thanks!

**Pull request merged**